tests/test_callbacks.cpp

```diff
@@ -119,7 +119,7 @@ TEST_SUBMODULE(callbacks, m) {
 
     class AbstractBase {
     public:
-        virtual ~AbstractBase() = default;
+        virtual ~AbstractBase() {};
```

Comment:
This change is now causing clang-tidy errors.

Reply:
Nice :(. When the destructor is defaulted, the Intel compiler can't find it when linking (no clue why). In Debug builds only, RelWithDebInfo or Release builds are fine.

Reply:
We can add `// NOLINT` to silence the clang-tidy errors.

Reply:
I feel like I already had to do this for a poor compiler in a couple of places (Intel or PGI). Adding NOLINT is fine, and make it clear that it is needed for some reason (though I'd add a comment to explain why too).

Reply:
I added suggestions adding NOLINT and comments.